networkx/generators/tests/test_directed.py

```diff
@@ -112,6 +133,9 @@ def test_no_self_loops(self):
         G = random_uniform_k_out_graph(n, k, self_loops=False)
         assert nx.number_of_selfloops(G) == 0
         assert all(d == k for v, d in G.out_degree())
+        G = random_uniform_k_out_graph(n, k, self_loops=False, with_replacement=False)
+        assert nx.number_of_selfloops(G) == 0
+        assert all(d == k for v, d in G.out_degree())
```

Comment:
This would be another good target for test parametrization.

Also, since I think the goal here is to improve test coverage, it may make sense to add another test the explicitly probes the `with_replacement` behavior. Note that the type of the graph is expected to change based on this parameter!